### DISCUSSION TITLE
Make Pagination render more customizable

### DISCUSSION THREAD

**Discussion by amitaibu** [digitallyinduced/ihp#1123] [General]
Currently [renderPagination](https://github.com/digitallyinduced/ihp/blob/bee61ae756613ca4d09ba9960484d4e9fefe8fcc/IHP/Pagination/ViewFunctions.hs#L27) is quite hardcoded in it's look and functionality. We have to copy quite a lot of code in order to override it, to change HTML (e.g. to use Tailwind classes, or to remove the Items per page select list).

![Selection_999(307)](https://user-images.githubusercontent.com/125707/136665168-e5e4851e-b580-4e92-b24c-6c05deab79f3.jpg)

Maybe a better option would be to convert it to something more costumizable like [TableViewable](https://github.com/digitallyinduced/ihp/blob/18f104da69c526ff9e8ad3a6cdaedc6d39afb38c/IHP/Job/Dashboard/Types.hs#L35) - is that the right approach?

**Comment by mpscholten** [comment]
Yes agree 👍  Likely we could integrate this into the [CSSFramework](https://github.com/digitallyinduced/ihp/blob/master/IHP/View/CSSFramework.hs) module

**Reply by amitaibu** [reply]
For learning experience I was trying to look into it by trying first to change the types, but I suspect I'm not in the correct path :disappointed_relieved:  I didn't open a PR of [this](https://github.com/digitallyinduced/ihp/compare/master...amitaibu:pagintion-cssframework?expand=1) yet as it's not even in a draft mode :smile_cat: .

Some guidance on the types would be great - and I'll try to take it from there.

**Reply by mpscholten** [reply]
For the start I'd merge the `PaginationView` data structure into the `CSSFramework` data structure. That makes it simpler for the start. Then you could basically copy all the existing code from the existing call-sites of the `CSSFramework` functions.

**Reply by amitaibu** [reply]
Thanks - I've made some progress and was able to make it compile. I got a couple of questions I've added as comments on the PR - the important one to proceed is https://github.com/digitallyinduced/ihp/pull/1157/files/12fac8b717f899ddb208bf518528bf243a84436b#r732600163
